.github/actions/test-report/action.yml

```diff
@@ -59,8 +60,41 @@ inputs:
 runs:
   using: "composite"
   steps:
+    # If go-json reporter then it's required to transform to junit
+    - name: go-json - Download artifacts
+      if: contains('go-json', inputs.reporter)
+      uses: actions/download-artifact@v3
+      with:
+        name: ${{ inputs.artifact }}
+
+    - name: go-json - Setup golang
+      if: contains('go-json', inputs.reporter)
+      uses: actions/setup-go@v3
+      with:
+        go-version: "1.18"
+
+    - name: go-json - Transform go-json to java-junit
+      if: contains('go-json', inputs.reporter)
+      run: |
+        go install gotest.tools/gotestsum@latest
+        cat ${{ inputs.path }} | gotestsum --junitfile go-json-test-results.xml
+
+    - name: go-json - Report test results
+      if: contains('go-json', inputs.reporter)
+      uses: dorny/test-reporter@c9b3d0e2bd2a4e96aaf424dbaa31c46b42318226
+      with:
+        name: ${{ inputs.name }}
+        path: "go-json-test-results.xml"
+        reporter: 'java-junit'
+        list-suites: ${{ inputs.list-suites }}
+        list-tests: ${{ inputs.list-tests }}
+        fail-on-error: ${{ inputs.fail-on-error }}
+        only-summary: ${{ inputs.only-summary }}
+
+    # If no go-json reporter then as usual
```

Comment:
sounds like a new different action, it uses parameters to choose one action or another, makes things complex, and can cause side effects on projects that do not use Go.

Reply:
I'll refactor this in a different action, so the current action will call the other one then

Reply:
I understand that the goal is to use always the same pattern in all projects, it is fine, in that case I preferred to have the actions `.github/actions/test-report-go`, `.github/actions/test-report-base`, and one to join both `.github/actions/test-report` that has only the conditionals

Reply:
See now, I'll contribute to the upstream project, and likely we will move away from this implementation.